iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/MultiplexingClientTests.java

```diff
@@ -474,33 +474,6 @@ public void sendMessagesMaxDevicesAllowedTimes10MultiplexingClientsSerialOpen()
         log.debug("Close time: " + (finishCloseTime - startCloseTime) / 1000.0);
     }
 
-    @ContinuousIntegrationTest
-    @Test
-    public void sendMessagesMaxDevicesAllowedRegisterAfterOpen() throws Exception
```

Comment:
See the description of this PR for more details on why this is being removed

Reply:
By removing this test, we also save 1500 test identities from creation. In total, our e2e tests use about 4500 identities, so this would bring us down to 3000. The identity recycling logic I added earlier helps, but there is no guarantee that all 1500 over these test identities would be recycled identities or that they will be recycled later either.

Reply:
The other 3000 identities can be broken into 1500 for the ```sendMessagesMaxDevicesAllowed()``` test that makes this deleted test redundant, and the rest of the 1500 are scattered among our fault injection tests, twin tests, method tests, telemetry tests, file upload tests, etc...

Reply:
One interesting discussion we may want to have in the future is if we want to disable the ```sendMessagesMaxDevicesAllowed()``` test as well. Our .NET SDK doesn't have a comparable test, and removing it would save another 1500 device identities.